src/utils/validation.py

```
@@ -0,0 +1,685 @@
+
+        except (ValueError, TypeError):
+            raise SecurityValidationError(f"Invalid numeric value for {field_name}")
+
+    @staticmethod
+    def validate_range(value: Any, min_val: Optional[float] = None,
+                      max_val: Optional[float] = None, field_name: str = "value") -> Union[int, float]:
+        """Validate numeric value within specified range"""
+        num = SecureNumeric.validate_positive_number(value, field_name)
+
+        if min_val is not None and num < min_val:
+            raise SecurityValidationError(f"{field_name} must be >= {min_val}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)